packages/sdk-middleware-auth/src/build-requests.js

```diff
@@ -43,8 +46,47 @@ export function buildRequestForClientCredentialsFlow (
   return { basicAuth, url, body }
 }
 
-export function buildRequestForPasswordFlow () {
-  // TODO
+export function buildRequestForPasswordFlow (
+  options: PasswordAuthMiddlewareOptions,
+): BuiltRequestParams {
+  if (!options)
+    throw new Error('Missing required options')
+
+  if (!options.host)
+    throw new Error('Missing required option (host)')
+
+  if (!options.projectKey)
+    throw new Error('Missing required option (projectKey)')
+
+  if (!options.credentials)
+    throw new Error('Missing required option (credentials)')
+
+  const {
+    clientId,
+    clientSecret,
+    user,
+  } = options.credentials
+  const pKey = options.projectKey
```

Comment:
shorter var names to avoid eslint error and unnecessary multiline

I think it's a good trade off 🤷‍♂️